---
author: Elizabeth Czarniak (CZARNIA_ELIZ@bentley.edu)
---

Let's assume that you already made your multivariate linear model,
similar to the one shown below. If you still need to create one, first see
how to fit a multivariate linear model.

We use example data here, but you would use your own data instead.

In [1]:
import pandas as pd
import statsmodels.api as sm
data = {
    'x1' : [ 2,  7,  4,  3, 11, 18,   6, 15,   9,  12],
    'x2' : [ 4,  6, 10,  1, 18, 11,   8, 20,  16,  13],
    'x3' : [11, 16, 20,  6, 14,  8,   5, 23,  13,  10],
    'y' :  [24, 60, 32, 29, 90, 45, 130, 76, 100, 120]
}

The following code fits the model to the data.

In [2]:
df = pd.DataFrame(data)
xs = df[['x1', 'x2', 'x3']]
y = df['y']
xs = sm.add_constant(xs)
model = sm.OLS(y, xs).fit()

/opt/conda/lib/python3.9/site-packages/statsmodels/tsa/tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


Now we want to test whether the model is significant.  We will use a null hypothesis
that states that all of the model's coefficients are equal to zero, that is, they are
not jointly significant in predicting $y$.  We can write $H_0: \beta_0 = \beta_1 = \beta2 = \beta_3 = 0$.

We also choose a value $0 \le \alpha \le 1$ as our Type 1 error rate. Herer we'll use
$\alpha=0.05$.

The summary output for the model will give us both the F-statistic and the p-value.

In [3]:
model.summary()

/opt/conda/lib/python3.9/site-packages/scipy/stats/stats.py:1541: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=10
  warnings.warn("kurtosistest only valid for n>=20 ... continuing "


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.594
Model:                            OLS   Adj. R-squared:                  0.390
Method:                 Least Squares   F-statistic:                     2.921
Date:                Tue, 05 Oct 2021   Prob (F-statistic):              0.122
Time:                        19:14:39   Log-Likelihood:                -45.689
No. Observations:                  10   AIC:                             99.38
Df Residuals:                       6   BIC:                             100.6
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         77.2443     27.366      2.823      0.030      10.282     144.206
x1            -2.7009      2.855     -0.946      0.381      -9.686       4.284
x2             7.2989      2.875      2.539      0.044       0.265      14.333
x3            -4.8607      2.187     -2.223      0.068     -10.211       0.490
==============================================================================
Omnibus:                        2.691   Durbin-Watson:                   2.123
Prob(Omnibus):                  0.260   Jarque-Bera (JB):                1.251
Skew:                           0.524   Prob(JB):                        0.535
Kurtosis:                       1.620   Cond. No.                         58.2
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Near the top right of the output, we can see that the F-statistic is 2.921.
The corresponding $p$-value immediately below it is 0.1222, which is greater than $\alpha$,
so we do not have sufficient evidence to reject the null hypothesis.

We cannot conclude that the independent variables in our model are jointly significant
in predicting the response variable.